xarray/core/dataarray.py

```diff
@@ -1519,12 +1519,68 @@ def interp(
 
         Examples
         --------
-        >>> da = xr.DataArray([1, 3], [("x", np.arange(2))])
-        >>> da.interp(x=0.5)
-        <xarray.DataArray ()>
-        array(2.)
+        da = xr.DataArray(
+            data=[
+                [0.1, 4.3, 2.1, 9.8],
+                [2.4, 7.9, 0.6, np.nan],
+                [6.9, np.nan, 5.3, 8.7],
+            ],
+            dims=("x", "y"),
+            coords={"x": [0, 1, 2], "y": [10, 12, 14, 16]},
+        )
+        >>> da
+        <xarray.DataArray (x: 3, y: 4)>
+        array([[0.1, 4.3, 2.1, 9.8],
+               [2.4, 7.9, 0.6, nan],
+               [6.9, nan, 5.3, 8.7]])
         Coordinates:
-            x        float64 0.5
+          * x        (x) int64 0 1 2
+          * y        (y) int64 10 12 14 16
+
+        1D linear interpolation (the default)
+        >>> da.interp(x=[0.25, 0.75, 1.25, 1.75])
```

Comment:
Very low confidence suggestion, please ignore unless you positively agree with it:

How about making one of these match the original value, so it's easy to link back to the original example?

```suggestion
        >>> da.interp(x=[0.25, 1., 1.25, 1.75])
```